### ISSUE TITLE
Networking: Advanced Networking - Multiple Interfaces

### ISSUE THREAD

**Issue by mcastelino** [01org/cc-oci-runtime#94]
The current implementation supports only a single network and only IPv4. This is due to the limitation of the kernel command line based on https://www.kernel.org/doc/Documentation/filesystems/nfs/nfsroot.txt
This allows for a single interface and IP.

As next step we need to support
- Support multiple networks (i.e interfaces)

This requires cloud-init or alternate method of sending in the network configuration and actually configuration of the same within the VM.

This issues will track the architecture and progress of the implementation

**Comment by jodh-intel** [comment]
Could you raise a PR to add further details (as you've outlined here) of the networking limitations:

https://github.com/01org/cc-oci-runtime/blob/master/README.rst#networking

**Comment by dlespiau** [comment]
List of sub-tasks:

- [ ] Provide a more flexible way to build the list of qemu arguments https://github.com/01org/cc-oci-runtime/issues/219
- [ ] Propagate and setup the full network config to the VM https://github.com/01org/cc-oci-runtime/issues/220

**Comment by dlespiau** [comment]
Split up the IPv6 part into its own issue: https://github.com/01org/cc-oci-runtime/issues/247

**Comment by jodh-intel** [comment]
This is blocked on hyperstart (#271).

**Comment by amshinde** [comment]
Fix added in 2.1 branch. QA can verify and close this.

**Comment by chavafg** [comment]
@mcastelino, @GabyCT added some tests on #524 to verify multiple interfaces, should we close this one? or do you think we are missing something?

**Comment by mcastelino** [comment]
@chavafg #524 covers all the tests we need. It covers both multiple interfaces. And multiple IPs per interface
